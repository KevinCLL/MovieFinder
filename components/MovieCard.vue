<script>
export default {
  props: {
    movie: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {
      expandedMovie: null,
      posterNotFound:
        'https://i.pinimg.com/originals/c2/d1/e8/c2d1e8ba4e5b23bbad9bf78b695da39a.jpg',
    }
  },
  async mounted() {
    const expandedMovie = await this.$axios.$get('http://www.omdbapi.com/', {
      params: {
        apikey: '89fef3ea',
        i: this.movie.imdbID,
        plot: 'full',
      },
    })
    this.expandedMovie = expandedMovie
  },
}
</script>

<template>
  <div class="flex flex-col items-center w-full p-3 h-full">
    <div class="border bg-white p-2 h-moviecard relative">
      <div
        v-if="movie.Poster !== 'N/A'"
        class="mb-2 flex items-center h-poster"
      >
        <img class="h-100" :src="movie.Poster" :alt="movie.Title" />
      </div>
      <div
        v-else
        class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-500 to-indigo-700 flex"
      >
        <p class="text-3xl text-white ml-10">Poster not found</p>
      </div>
      <div class="h-5">
        <h2 class="text-2xl">
          <span class="bold">{{ `${movie.Title}` }}</span>
          <span
            v-if="expandedMovie && expandedMovie.Director !== 'N/A'"
            class="italic"
          >
            {{ `by ${expandedMovie.Director}` }}
          </span>
        </h2>
        <h3 class="text-xl">
          {{ movie.Year }}
        </h3>
      </div>
      <img
        class="cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
        src="~/assets/img/magnifying_glass_icon.png"
        alt="magnifying glass"
        @click="$emit('movie-select', expandedMovie)"
      />
    </div>
  </div>
</template>

<style scoped></style>
