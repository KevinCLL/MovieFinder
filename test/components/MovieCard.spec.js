/* eslint-disable no-console */
import { shallowMount } from '@vue/test-utils'
import MovieCard from '@/components/MovieCard.vue'
import detailedMovie from '../data/detailedMovie.js'
import cardMovie from '../data/cardMovie.js'

describe('MovieCard', () => {
  let wrapper
  let spy

  beforeEach(() => {
    spy = jest.spyOn(console, 'warn').mockImplementation()
    wrapper = shallowMount(MovieCard, {
      mocks: {
        $axios: {
          $get: () => Promise.resolve(detailedMovie),
        },
      },
      propsData: {
        movie: cardMovie,
      },
      data() {
        return {
          expandedMovie: null,
        }
      },
    })
  })
  afterEach(() => {
    if (wrapper) {
      wrapper.destroy()
    }
    spy.mockRestore()
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it("ExpandedMovie didn't incorrectly fills", () => {
    const failWrapper = shallowMount(MovieCard, {
      mocks: {
        $axios: {
          $get: () => Promise.reject(new Error('Fake error')),
        },
      },
      propsData: {
        movie: cardMovie,
      },
      data() {
        return {
          expandedMovie: null,
        }
      },
    })
    expect(failWrapper.vm.$data.expandedMovie).toBe(null)
    failWrapper.destroy()
  })

  it('ExpandedMovie fills', () => {
    expect(wrapper.vm.$data.expandedMovie).toBe(detailedMovie)
    expect(spy).not.toBeCalled()
  })
})
