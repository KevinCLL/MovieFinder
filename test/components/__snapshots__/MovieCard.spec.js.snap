// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<div
  class="flex flex-col items-center w-full p-3 h-full"
>
  <div
    class="border bg-white p-2 h-moviecard w-moviecard relative"
  >
    <div
      class="mb-2 flex h-poster"
    >
      <img
        alt="Primer"
        class="h-100 my-0 mx-auto"
        src="https://m.media-amazon.com/images/M/MV5BNjc3OWVjMTItYjc0Yi00NmFlLTk2YTgtYmU0MzcxMjBkNTYxXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
      />
    </div>
     
    <div
      class="h-5 text-center"
    >
      <h2>
        <span
          class="text-2xl"
        >
          Primer
        </span>
         
        <span
          class="italic text-xl"
        >
          
          by Shane Carruth
        
        </span>
      </h2>
       
      <h3
        class="text-xl"
      >
        
        2004
      
      </h3>
    </div>
     
    <img
      alt="magnifying glass"
      class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
      src="~/assets/img/magnifying_glass_icon.png"
    />
  </div>
</div>
`;
